spyder/plugins/ipythonconsole/confpage.py

```diff
@@ -255,6 +238,69 @@ def setup_page(self):
         run_file_group.setLayout(run_file_layout)
 
+        # ---- Debug ----
+        # Pdb run lines Group
+        pdb_run_lines_group = QGroupBox(_("Run code while debugging"))
+        pdb_run_lines_label = QLabel(_(
+            "You can run several lines of code on each "
+            "new prompt while debugging. Please "
+            "introduce each one separated by semicolons "
+            "and a space, for example:<br>"
+            "<i>import matplotlib.pyplot as plt</i>"))
+        pdb_run_lines_label.setWordWrap(True)
+        pdb_run_lines_edit = self.create_lineedit(
+            _("Lines:"), 'startup/pdb_run_lines', '', alignment=Qt.Horizontal)
+
+        pdb_run_lines_layout = QVBoxLayout()
+        pdb_run_lines_layout.addWidget(pdb_run_lines_label)
+        pdb_run_lines_layout.addWidget(pdb_run_lines_edit)
+        pdb_run_lines_group.setLayout(pdb_run_lines_layout)
+
+        # Debug Group
+        debug_group = QGroupBox(_("Debug"))
+        debug_layout = QVBoxLayout()
+
+        prevent_closing_box = newcb(
+            _("Prevent editor from closing files while debugging"),
+            'pdb_prevent_closing',
+            tip=_("This option prevents the user from closing a file while"
+                  " it is debugged."))
+        debug_layout.addWidget(prevent_closing_box)
+
+        continue_box = newcb(
+            _("Stop debugging on first line of files without breakpoints"), 'pdb_stop_first_line',
+            tip=_("This option lets you decide if the debugger should"
+                  " stop on the first line while debugging if no breakpoints"
+                  " are present."))
+        debug_layout.addWidget(continue_box)
+
+        libraries_box = newcb(
+            _("Ignore Python libraries while debugging"), 'pdb_ignore_lib',
+            tip=_("This option lets you decide if the debugger should "
+                  "ignore the system libraries while debugging."))
+        debug_layout.addWidget(libraries_box)
+
+        execute_events_box = newcb(
+            _("Process execute events while debugging"), 'pdb_execute_events',
+            tip=_("This option lets you decide if the debugger should "
+                  "process the 'execute events' after each prompt, such as "
+                  "matplotlib 'show'."))
+        debug_layout.addWidget(execute_events_box)
+
+        exclamation_mark_box = newcb(
+            _("Use exclamation mark prefix for Pdb commands"), 'pdb_use_exclamation_mark',
+            tip=_("This option lets you decide if the Pdb commands should "
+                  "be prefixed by an excalamtion mark. This helps in separating Pdb "
+                  "commands from python code."))
+        debug_layout.addWidget(exclamation_mark_box)
+
+        single_letter_enter_box = newcb(
+            _("Use enter to leave explore mode."), 'pdb_single_letter_enter',
+            tip=_("This option lets you stop the explore mode by pressing enter."))
```

Comment:
I really don't understand the idea behind this new "Explore mode". Is "Explore mode" not the same as Pdb without exclamations?

Reply:
So the idea is that when you follow the code execution, you want to be as fast as possible because you might need to step through a lot of code to understand what is happening. Everything I can do to reduce the amount of typing is good in this situation. This is why Pdb has single letter commands. The "Explore mode" removes the need to use "Enter", which divides by two the number of keystrokes needed compared to regular pdb (by 3 if you have to add "!")

Reply:
> Is "Explore mode" not the same as Pdb without exclamations?

No because the only thing you can do is the pdb single letter commands. The ones I use most are:
 - (n)ext
 - (s)tep
 - (r)eturn
 - (c)ontinue
 - (u)p
 - (d)own
 - (w)here

The first 4 have a shortcut but I find it more intuitive to use the keyboard key associated with the command rather than a F11-based shortcut